app/js/components/manageApps/ManageApps.jsx

```diff
@@ -243,7 +249,26 @@ export default class ManageApps extends React.Component {
 
   handleDownload(e) {
     e.preventDefault();
-    location.href=this.state.downloadUri;
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', this.state.downloadUri, true);
+    xhr.responseType = 'blob';
+    let filename = this.state.downloadUri.split('=').slice(-1)[0];
+    let that = this;
+    
+    xhr.onload = function(e) {
+      if (this.status == 200) {
+        let data = this.response;
```

Comment:
Just a personal thought, I am a bit confused where `this.response` is coming from. Is it from the `local scope of xhr.onload`? If yes: is there any reason why we cant use a promise here to make it more readable. Thanks.

Reply:
I tried using promises with an ajax and axios call but somehow the file type/signature was always compromised as the file is processed as string. But feel free to try out using promises, I will be happy to have more readable code.

Reply:
Oh, makes sense then. good job @paulupendo